### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**
[DEMO LINK](https://Zelinskyi-Serhii.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

**Review by rashyd-hasratov — APPROVED** [review]
Good job!

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,118 @@
...
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortType, setType] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
```
Comment (sergii-nosachenko) [inline src/App.tsx:50]:
```suggestion
  const [sortType, setSortType] = useState(SortType.NONE);
  const [isReversed, setIsReversed] = useState(false);
```

https://react.dev/reference/react/useState#usestate

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,118 @@
...
-          Reset
-        </button>
+        {showResetButton && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setType(SortType.NONE);
+              setReverse(false);
+            }}
```
Comment (sergii-nosachenko) [inline src/App.tsx:104]: Create handler for reset action

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,118 @@
...
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {goods.map(good => (
+            <Good good={good} key={uuidv4()} />))}
         </ul>
       </ul>
```
Comment (sergii-nosachenko) [inline src/App.tsx:117]: Create a separate component for the goods list

File: src/Good.tsx
Comment (sergii-nosachenko) [inline src/Good.tsx:1]: Better convert this file to GoodsList

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,62 +1,117 @@
...
+}
 
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setSortReverse] = useState(false);
+
+  const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+  const showResetButton = (sortType !== SortType.NONE || isReversed);
+
+  const HeandlerReset = () => {
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:54]:
```suggestion
  const handleReset = () => {
```

Comment (OleksiiBlezniuk) [reply src/App.tsx:54]: 1 - we use uppercased names for classes, components, but not for everything
2 - function name should represent action

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**